src/server/middleware/createRequestHtmlFragment.jsx

```diff
@@ -89,11 +94,16 @@ export default function createRequestHtmlFragment({ createRoutes }) {
       if (getRenderMethodName(state) === 'renderForStaticMarkup') {
         await dispatch(composeModules(routeModules));
       } else {
-        const fallback = await getModuleDataBreaker.fire({
+        const { fallback, redirect } = await getModuleDataBreaker.fire({
           dispatch,
           modules: routeModules,
         });
 
+        if (redirect) {
+          res.redirect(redirect.status, redirect.url);
```

Comment:
Should we default the status to `302` (express default)
```suggestion
          res.redirect(redirect.status || 302, redirect.url);
```